src/__experimental__/components/input/input-presentation.spec.js

```diff
@@ -104,73 +105,50 @@ describe('InputPresentation', () => {
-    });
-
-    it('provides hasFocus state defaulting to false', () => {
-      expect(context.hasFocus).toEqual(false);
-    });
+    let wrapper;
 
-    it('enables focus on focus', () => {
-      expect(context.hasFocus).toEqual(false);
-      context.onFocus();
-      context = getContext(wrapper);
-      expect(context.hasFocus).toEqual(true);
-    });
+    beforeEach(() => {
+      contextOnMouseEnter = jest.fn();
+      groupContextOnMouseEnter = jest.fn();
 
-    it('disables focus on blur', () => {
-      expect(context.hasFocus).toEqual(true);
-      context.onBlur();
-      context = getContext(wrapper);
-      expect(context.hasFocus).toEqual(false);
-    });
+      contextOnMouseLeave = jest.fn();
+      groupContextOnMouseLeave = jest.fn();
 
-    it('provides hasMouseOver state defaulting to false', () => {
-      expect(context.hasMouseOver).toEqual(false);
+      wrapper = renderWithContext({},
+        {
+          onMouseEnter: groupContextOnMouseEnter,
+          onMouseLeave: groupContextOnMouseLeave
+        },
+        {
+          onMouseEnter: contextOnMouseEnter,
+          onMouseLeave: contextOnMouseLeave
+        },);
     });
 
-    it('enables hasMouseOver on mouse enter', () => {
-      expect(context.hasMouseOver).toEqual(false);
-      wrapper.find(InputPresentationStyle).simulate('mouseenter');
-      context = getContext(wrapper);
-      expect(context.hasMouseOver).toEqual(true);
+    it('triggers onMouseEnter callbacks passed from context', () => {
+      wrapper.find(InputPresentationStyle).props().onMouseEnter();
+      expect(contextOnMouseEnter).toHaveBeenCalled();
+      expect(groupContextOnMouseEnter).toHaveBeenCalled();
     });
 
-    it('disables hasMouseOver on mouse leave', () => {
-      expect(context.hasMouseOver).toEqual(true);
-      wrapper.find(InputPresentationStyle).simulate('mouseleave');
-      context = getContext(wrapper);
-      expect(context.hasMouseOver).toEqual(false);
+    it('triggers onMouseLeave callbacks passed from context', () => {
+      wrapper.find(InputPresentationStyle).props().onMouseLeave();
+      expect(contextOnMouseLeave).toHaveBeenCalled();
+      expect(groupContextOnMouseLeave).toHaveBeenCalled();
     });
 
-    it('assigns a given input to the component', () => {
-      expect(wrapper.instance().input).toEqual({});
-      context.inputRef({ current: 'my input!' });
-      expect(wrapper.instance().input).toEqual({ current: 'my input!' });
-    });
-
-    describe('on mouse down of the wrapping div', () => {
-      it('focuses on the input after a 0 timeout', () => {
-        jest.useFakeTimers();
-        const focus = jest.fn();
-        context.inputRef({ current: { focus } });
-        wrapper.simulate('mousedown');
-        expect(focus).not.toHaveBeenCalled();
-        jest.runAllTimers();
-        expect(focus).toHaveBeenCalled();
-      });
+    it('nothing happens if onFocus, onBlur, onMouseEnter or onMouseLeave callbacks are not provided', () => {
```

Comment:
Same here. I think something like `it('does nothing if onFocus, onBlur, onMouseEnter or onMouseLeave callbacks are not provided'` would be better.